README.md

```diff
@@ -0,0 +1,492 @@
+[![](https://travis-ci.org/buildo/avenger.svg)](https://travis-ci.org/buildo/avenger)
+[![](https://img.shields.io/npm/v/avenger.svg?sytle=flat-square)](https://www.npmjs.com/package/avenger)
+[![npm downloads](https://img.shields.io/npm/dm/avenger.svg?style=flat-square)](https://www.npmjs.com/package/avenger)
+[![](https://david-dm.org/buildo/avenger.svg)](https://david-dm.org/buildo/avenger#info=dependencies&view=list)
+[![](https://david-dm.org/buildo/avenger/dev-status.svg)](https://david-dm.org/buildo/avenger#info=devDependencies&view=list)
+
+---
+
+# Intro
+
+Avenger is a data fetching and caching layer written in TypeScript. Its API is designed to mirror the principles of **Command Query Responsibility Segregation** and facilitate their adoption (if you are new to the concept you can get a grasp of its foundations in [this nice article](https://martinfowler.com/bliki/CQRS.html) by Martin Fowler).
+
+Building user interfaces is a complex task, mainly because of its `IO` intensive nature. Reads (**queries**) and updates (**commands**) toward "external" data sources are ubiquitous and difficult to orchestrate but _orchestration_ is not the only challenge a UI developer faces, _performance_ and _scalability_ are also key aspects of good design.
+
+We believe that an _effective and powerful abstraction to handle caching and synchronization of external data in a declarative way_ is of fundamental importance when designing a solid user interface.
+
+This is what **Avenger** wants to be: an abstraction layer over external data that handles caching and synchronization for you:
+
+!["cached flow"](docs/Avenger.svg)
+
+By separating how we fetch external data and how we update it we are able to state in a very declarative and _natural_ way the correct lifecycle of that data:
+
+```ts
+// define a cached query, with strategy "available" (more about this later)
+const user = queryShallow((id: string) => API.fetchUser(id), available);
```

Comment:
I know this is probably my own example copy-pasted, but I'd put `queryStrict` here, since the input is a plain primitive value (`string`)